### PR TITLE
docs(spec): pin deterministic Bun toolchain for GH1686

### PR DESCRIPTION
**Pull request by majiayu000**
Linked work: #1686

## Why

PR #1685 exposed a CI reliability defect unrelated to its Rust-only diff: two exact-head attempts failed inside `oven-sh/setup-bun@v2` before any project command ran because `bun-version: latest` enumerated the Bun tag collection and GitHub returned HTTP 503. The exact web build passed locally.

The upstream action source confirms that a strict semantic version constructs the release URL directly, while `latest` and ranges request the full tags endpoint. Harness currently repeats `latest` in three active workflows.

## Specification plan

1. Establish root `.bun-version` as the single source of truth at Bun 1.3.14.
2. Point all three active setup-bun steps to that file.
3. Preserve every existing install, typecheck, test, build, artifact, and Rust validation command.
4. Verify exact resolver behavior, frozen web dependencies, workflow scope, CI, review threads, and SpecRail gates.

## Contents

- `product.md`: user problem, goals, non-goals, B-001 through B-008, edge cases, rollout.
- `tech.md`: exact root cause, upstream resolver behavior, four-file design, invariants, alternatives, risks, test and rollback plans.
- `tasks.md`: deterministic baseline, implementation, verification, PR gate, and dependent-PR recovery steps.

## Verification

- `python3 checks/check_workflow.py --repo . --spec-dir specs/GH1686`
- `python3 checks/route_gate.py --repo . --route write_spec --issue 1686 --state ready_to_spec --label ready_to_spec --artifact product_spec=specs/GH1686/product.md --artifact tech_spec=specs/GH1686/tech.md --mode advisory --json`
- `cargo fmt --all -- --check`
- `cargo clippy --workspace --all-targets -- -D warnings`
- Pre-push database-independent workspace library tests: 482 passed; 6 PostgreSQL lease tests deferred because this documentation-only worktree had no database URL.
- Manual VibeGuard L1-L7 review: passed; no automated VibeGuard CLI is installed.

## Scope

Specification only. This PR does not change workflow, toolchain, dependency, web, SDK, or Rust behavior. Implementation follows in a separate PR after this packet merges.

### CODE REVIEW

File: specs/GH1686/product.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      `oven-sh/setup-bun@v2` through `bun-version-file: .bun-version`.
+- [ ] B-003: No active workflow retains `bun-version: latest` or duplicates a
+      literal Bun version.
+- [ ] B-004: The strict version path constructs the release download without
+      requesting `https://api.github.com/repos/oven-sh/bun/git/refs/tags`.
+- [ ] B-005: Bun 1.3.14 completes frozen dependency installation, web
+      typecheck, tests, SDK prebuild, and production build without manifest or
+      lockfile changes.
+- [ ] B-006: Existing workflow triggers, job dependencies, commands, artifact
+      handling, and Rust validation behavior remain unchanged.
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply specs/GH1686/product.md]: Addressed in 25939ac9. The spec now requires `.bun-version` in the main CI `ci` filter and Web CI path triggers, explicitly preserves only unrelated trigger behavior, and adds path-classification verification. The issue acceptance criteria were updated to match.
